Do not force optimization flags on user

bicatali@gentoo.org

--- configure.ac.orig	2012-05-29 23:16:20.000000000 +0100
+++ configure.ac	2012-05-29 23:20:48.000000000 +0100
@@ -773,8 +773,6 @@
 if test x$enable_debug = xyes; then
 	CFLAGS="$CFLAGS -O0"
 	AC_DEFINE(STARPU_SPINLOCK_CHECK, [1], [check spinlock use])
-else
-	CFLAGS="$CFLAGS -O3"
 fi
 CFLAGS+=" -gdwarf-2 -g3 "
 
